cranelift/isle/isle/src/sema.rs

```diff
@@ -82,26 +83,144 @@ pub struct TypeEnv {
 
 /// A built-in type.
 #[derive(Copy, Clone, Debug, PartialEq, Eq)]
+#[repr(u8)]
 pub enum BuiltinType {
     /// The type of booleans, with values `true` and `false`.
     Bool,
+    /// The types of fixed-width integers.
+    Int(IntType),
+}
+
+/// A built-in fixed-width integer type.
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum IntType {
+    /// Unsigned, 8 bits.
+    U8,
+    /// Unsigned, 16 bits.
+    U16,
+    /// Unsigned, 32 bits.
+    U32,
+    /// Unsigned, 64 bits.
+    U64,
+    /// Unsigned, 128 bits.
+    U128,
+    /// Unsigned, enough bits to hold a pointer.
+    USize,
+    /// Signed, 8 bits.
+    I8,
+    /// Signed, 16 bits.
+    I16,
+    /// Signed, 32 bits.
+    I32,
+    /// Signed, 64 bits.
+    I64,
+    /// Signed, 128 bits.
+    I128,
+    /// Unsigned, enough bits to hold a pointer.
+    ISize,
+}
+
+impl IntType {
+    /// Get the integer type's name.
+    pub fn name(&self) -> &'static str {
+        match self {
+            IntType::U8 => "u8",
+            IntType::U16 => "u16",
+            IntType::U32 => "u32",
+            IntType::U64 => "u64",
+            IntType::U128 => "u128",
+            IntType::USize => "usize",
+            IntType::I8 => "i8",
+            IntType::I16 => "i16",
+            IntType::I32 => "i32",
+            IntType::I64 => "i64",
+            IntType::I128 => "i128",
+            IntType::ISize => "isize",
+        }
+    }
+
+    /// Is this integer type signed?
+    pub fn is_signed(&self) -> bool {
+        matches!(
```

Comment:
Can we write a `match` with explicit arms for all types here? I'd prefer that so we are forced to think about any new types we add explicitly. The pattern as written makes that easy to miss...